weblate/trans/models/translation.py

```diff
@@ -1473,6 +1485,7 @@ def delete_unit(self, request, unit):
                     source_unit = translation_unit.source_unit
                     if source_unit.source_unit.unit_set.count() == 1:
                         source_unit.delete()
+                        self.notify_deletion(source_unit, user)
```

Comment:
This is the correct Translation object to trigger notification:

```suggestion
                        source_unit.translation.notify_deletion(source_unit, user)
```

Reply:
I am not so sure that's correct. I have tested my code change locally and it works with weblate fedora messaging. I was also able to consume the message from a custom consumer too. Also the `notify_delete` method is part of the same class so `self.notify_delete` should be sufficient.

Unless there is some additional context I am unaware of?

Reply:
You are deleting the source unit here, so the matching Translation object should get the notification, not the current translation. Also, this code path is not used in monolingual setup, so you are probably not testing this at all ;-).

Reply:
Oh yeah valid point, this has been altered accordingly.